assets/resource.py

```diff
@@ -46,6 +47,26 @@ def check(self, source, version):
 
         return new_versions
 
+    def check(self, source, version):
+        """Check for new version(s)."""
+
+        index = source['index']
+        ssl_verify = source.get('ssl_verify', True)
+
+        if isinstance(ssl_verify, bool):
+            verify = ssl_verify
+        elif isinstance(ssl_verify, str):
+            verify = str(tempfile.NamedTemporaryFile(delete=False, prefix='ssl-').write(verify))
+
+        # request index
+        response = requests.request('GET', index, verify=verify)
+        response.raise_for_status()
+
+        if source.get('etag'):
+            return self.check_etag(source, version, response)
+        else:
```

Comment:
Makes sense. Unfortunately, both of these methods are a bit contrived, and I don't see a use case for fallback. The resource should fail if it lacks an etag. Sounds like I need another test or two. :)